### PR TITLE
docs: add axis annotation example

### PR DESCRIPTION
**Pull request by chrisprice**

### CODE REVIEW

File: examples/axis-annotation/index.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const annotation = new Annotation(45, 0.01);
+
+const scale = d3.scaleLinear().domain([0, 100]);
+
+const axis = fc
+    .axisBottom(scale)
+    .tickValues([...scale.ticks(), annotation])
+    .decorate(g => {
+        g.filter(d => d === annotation)
+            .select('path')
```
Comment (ColinEberhardt) [inline examples/axis-annotation/index.js:38]: should this use `enter()` to ensure that these style / attributes are only set when a new element is created?

File: examples/axis-annotation/index.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+class Annotation {
+    constructor(value, velocity) {
+        this.value = value;
+        this.velocity = velocity;
+    }
+
+    animate() {
+        this.velocity += (Math.random() - 0.5) * 0.01;
+        this.value += this.velocity;
+        if (this.value < scale.domain()[0]) {
```
Comment (ColinEberhardt) [inline examples/axis-annotation/index.js]: <minor/>

I find indexed values returned by accessors really hard to read `scale.domain()[0]`, perhaps pull out the scale range as a variable?
```
const range = [0, 100];
...
this.value = range[0] + (range[0] - this.value)
...
const scale = d3.scaleLinear().domain(range);
```

**Review by ColinEberhardt — APPROVED** [review]
Useful example, a couple of minor comments, I'll leave to your discretion.

File: examples/axis-annotation/index.js (hunk 1 above)
Comment (chrisprice) [reply examples/axis-annotation/index.js:38]: I could for the styles but the axis code sets the `d` attribute on each update so I figured I'd just lump them together for simplicity

**Pull request merged**